reactive/containerd.py

```diff
@@ -560,13 +566,25 @@ def configure_nvidia(reconfigure=True):
         )
         for source in sources
     ]
-    with open("/etc/apt/sources.list.d/nvidia.list", "w") as f:
+    with open(NVIDIA_SOURCES_FILE, "w") as f:
         f.write("\n".join(formatted_sources))
 
+
+@when("containerd.nvidia.available")
+@when_not("containerd.nvidia.ready", "endpoint.containerd.departed")
+def install_nvidia_drivers(reconfigure=True):
+    """Based on charm config, install and configure NVIDIA drivers.
+
+    :return: None
+    """
+    # Fist remove any existing nvidia drivers
+    unconfigure_nvidia(reconfigure=False)
+    configure_nvidia_sources()
+
     apt_update()
     nvidia_packages = config("nvidia_apt_packages").split()
     if not nvidia_packages:
-        status.blocked("No NVIDIA packages selected to install.")
+        set_state("containerd.nvidia.missing_package_list")
```

Comment:
here's another flag that if it's ever set -- will never be cleared.  consider clearing it after or not using it at all.  I don't know why the blocked status have been moved to the update-status hook.  it delays showing the user the charm is blocked until that hook runs -- which could be 5 minutes after the config changed